### DISCUSSION TITLE
Voice instructions and turn indicator giving incorrect directions

### DISCUSSION THREAD

**Discussion by daveshaw** [osmandapp/OsmAnd#13546] [Q&A]
I am new to OSMAND and am using Android version 4.1.11 OsmAnd's in-app offline routing. The map I am using is the England South West England dated 1/11/2021. I used the Cycling profile.

I used the "Plan Route" function to create a short cycle route with just 16 points. I then select the track, activate navigation, and start cycling. The arrow on the map is always correct as I travel. However after the first 2 turns the voice instruction and onscreen turn indicator start to misbehave, although they are in agreement with each other. Sometimes the voice control says turn left when the turn is clearly a right, Sometimes it says turn when there is no turn nearby. Sometimes it clearly thinks I am approaching a point at least 1km and another turn further on in the journey than I actually am. Sometimes it suggests crazy side-diversions onto offroad tracks. Sometimes it appears to get a grip again on where it should be routing but then loses the plot again.

I have tried the same cycle route twice in the last few days and many (or perhaps all?) of the voice control mistakes mistakes were consistent on both rides.

I attach screenshots which demonstrate some of the disconnects between the route I planned and the directions I receive. I also attack the track GPX exported from OSMAND.

Happy to help debug if you can guide me. Thanks.

<table style="width:100%">
   <tr>
<th><img src="https://user-images.githubusercontent.com/2145435/147855934-299e40ba-c9a9-4984-b492-a42ccd1ee465.png"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/2145435/147855935-4395335d-0b52-49a7-a497-eedda4c8e23e.png"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/2145435/147855936-effe639c-24d5-484a-a37b-aebbc8419d8e.png"  width="300" height="auto"/> </th>
 <th><img src="https://user-images.githubusercontent.com/2145435/147855931-8ed5650e-c808-482f-a867-a851c003e32e.png"  width="300" height="auto"/> </th>
  </tr>
</table> 

[Drum loop.gpx.txt](https://github.com/osmandapp/OsmAnd/files/7797962/Drum.loop.gpx.txt)

**Comment by daveshaw** [comment]
If it helps at all I did a ride (same area, same device etc. but not the same route) this morning where I just navigated to a marker and everything worked perfectly.

**Comment by daveshaw** [comment]
I have just created another nearby route using the Plan a Route function. Again when I attempted to navigate it everything went fine for the first 6 turns or so and then the voice control and turn indicator clearly lost track of me. If I had to guess I'd say that it goes wrong when they thought I'd made the next turn well before (150 metres?) before I had. From that point on it appeared to be directing me as if I were at least 1 instruction ahead of where I actually was.

**Comment by vshcherb** [comment]
Navigating by GPX is a complex subject on itself, it requires to attach to the roads if GPX was not created by OsmAnd. So OsmAnd can recognize all the turns. Voice alerts are configurable by one slider and you can see information when they should be announced in Voice Prompts settings

**Reply by daveshaw** [reply]
Hi @vshcherb , Thanks for responding. In my case the route was created within Osmand using the Plan a Route function not by importing a GPX.

**Comment by barteis** [comment]
You said that  you "just navigated to a marker and everything worked perfectly."  I wonder what happens if you navigate your entire route using a destination and intermediate destinations using OsmAnd's regular navigation menu (as opposed to Plan a Route.)  Do the problems persist?

**Reply by daveshaw** [reply]
Hi @barteis , this morning I created a bunch of markers along the same route and used them as intermediate destinations on the regular Navigation menu. Everything went perfectly.

What does this tell us? I'm happy to try any other tests you can suggest. I'm really hoping to get this fixed as I much prefer the way Plan a Route allows you to sequentially build a route from points, and it also doesn't treat the points as worth mentioning on the voice control whereas the markers (which have no significance other than as forming part of the route) trigger announcements.

**Reply by barteis** [reply]
It does sound like a bug.  Perhaps this is related to https://github.com/osmandapp/OsmAnd/issues/13031 ? I agree with you about Plan A Route: some real advantages.  But of course, the navigation has to work.

**Reply by daveshaw** [reply]
@barteis Yes, I think this is a duplicate of [https://github.com/osmandapp/OsmAnd/issues/13031] too.

I'm new to Osmand+ but disappointed that such a useful piece of functionality is broken, although it feels as though the fix might be quite simple. As a cyclist I want to use Plan a Route with intermediate points to ensure that my ride definitely includes specific roads/tracks - I rarely just want the best route between A and B.

**Reply by woheller69** [reply]
"As a cyclist I want to use Plan a Route with intermediate points to ensure that my ride definitely includes specific roads/tracks - I rarely just want the best route between A and B."

That is exactly what I need...

**Reply by barteis** [reply]
On https://github.com/osmandapp/OsmAnd/issues/13031, there's this: 
"@vshcherb vshcherb added this to the 4.3-backend milestone on Dec 1, 2021."
I assume that means a fix is coming, but as yet there's no due date. 

https://github.com/osmandapp/OsmAnd/milestone/31

**Comment by kosi2711** [comment]
Same for me - starting somewhere in 3.9.x. There has been a thread stating this, and it has been noted that it will be fixed in 4.0. Don't know if it has been fixed in 4.0.x, because I rarely used it. With 4.1 (already with 4.1.2-beta) the "run ahead" is definitly there.

My use case is: "plan a route" on a device running Android11 and Osmand 4.1.6 (the last usable release before the storage dilemma), transfer the gpx to a device running Android9 and Osmand 4.1.11. When navigating after some time the voice instructions run ahead like there would be some kind of simulation.

@barteis: >>_navigate your entire route using a destination and intermediate destinations using OsmAnd's regular navigation menu_
Doing so everything is correct.

2 annotations:
- POIs are correctly announed
- I have modified the (german) tts-files

Greets

**Reply by Zirochkabila** [reply]
We managed to find the cause (overlay route). Following your example, the bug report was created https://github.com/osmandapp/OsmAnd/issues/14331
